plugin/trino-influxdb/src/main/java/io/trino/plugin/influxdb/InfluxFilterManager.java

```diff
@@ -0,0 +1,259 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.trino.plugin.influxdb;
+
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import io.airlift.slice.Slice;
+import io.trino.spi.connector.ColumnHandle;
+import io.trino.spi.connector.Constraint;
+import io.trino.spi.expression.Call;
+import io.trino.spi.expression.ConnectorExpression;
+import io.trino.spi.expression.Constant;
+import io.trino.spi.expression.FunctionName;
+import io.trino.spi.expression.Variable;
+import io.trino.spi.predicate.Domain;
+import io.trino.spi.predicate.EquatableValueSet;
+import io.trino.spi.predicate.Range;
+import io.trino.spi.predicate.SortedRangeSet;
+import io.trino.spi.predicate.TupleDomain;
+import io.trino.spi.predicate.ValueSet;
+import io.trino.spi.type.BigintType;
+import io.trino.spi.type.DoubleType;
+import io.trino.spi.type.VarcharType;
+
+import java.math.BigDecimal;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+
+import static io.trino.spi.expression.StandardFunctions.AND_FUNCTION_NAME;
+import static io.trino.spi.expression.StandardFunctions.EQUAL_OPERATOR_FUNCTION_NAME;
+import static io.trino.spi.expression.StandardFunctions.NOT_EQUAL_OPERATOR_FUNCTION_NAME;
+import static io.trino.spi.expression.StandardFunctions.OR_FUNCTION_NAME;
+import static io.trino.spi.type.BooleanType.BOOLEAN;
+
+public class InfluxFilterManager
+{
+    private static final String AND = " AND ";
+    private static final String OR = " OR ";
+    private static final String GT = " > ";
+    private static final String GTE = " >= ";
+    private static final String LT = " < ";
+    private static final String LTE = " <= ";
+    private static final String EQ = " = ";
+    private static final String NEQ = " != ";
+    private static final String LPAREN = " ( ";
+    private static final String RPAREN = " ) ";
+
+    private static final Map<FunctionName, String> FUNCTION_MAP = ImmutableMap.of(
+            AND_FUNCTION_NAME, " AND ",
+            OR_FUNCTION_NAME, " OR ",
+            EQUAL_OPERATOR_FUNCTION_NAME, " = ",
+            NOT_EQUAL_OPERATOR_FUNCTION_NAME, " != ");
+
+    private InfluxFilterManager() {}
+
+    public static List<String> buildFilter(Constraint constraint)
+    {
+        TupleDomain<ColumnHandle> tupleDomain = constraint.getSummary();
+        ImmutableList.Builder<String> columnFilters = ImmutableList.builder();
+        tupleDomain.getDomains().ifPresent(columnHandleDomainMap -> {
+            columnHandleDomainMap.forEach((columnHandle, domain) -> {
+                String columnFilter = buildDomainFilter((InfluxColumnHandle) columnHandle, domain);
+                if (!Strings.isNullOrEmpty(columnFilter)) {
```

Comment:
Instead of checking if it is Null or empty - Can we use Optional or a different state like `TRUE` if the expression cannot be represented.